app-accessibility/caribou/files/caribou-fix-python-env.patch

```diff
@@ -0,0 +1,60 @@
+diff -rupN caribou-0.4.21.orig/bin/antler-keyboard.in caribou-0.4.21/bin/antler-keyboard.in
+--- a/bin/antler-keyboard.in	2016-07-20 01:13:11.000000000 -0400
++++ b/bin/antler-keyboard.in	2021-01-24 23:54:14.466466751 -0500
+@@ -18,24 +18,4 @@
+ # along with this program; if not, write to the Free Software Foundation,
+ # Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ 
+-script_dir="$(dirname "$(readlink -f ${0})")"
+-
+-prefix=@prefix@
+-exec_prefix=@exec_prefix@
+-datarootdir=@datarootdir@
+-
+-if [ $script_dir = "@libexecdir@" ]
+-then
+-  datadir="$(@PYTHON@ -c "from gi.repository import GLib; print(':'.join(GLib.get_system_data_dirs()))")"
+-  export PYTHONPATH="${prefix}/lib/python@PYTHON_VERSION@/site-packages:${prefix}/lib64/python@PYTHON_VERSION@/site-packages${PYTHONPATH:+:$PYTHONPATH}"
+-  export GI_TYPELIB_PATH="@libdir@/girepository-1.0${GI_TYPELIB_PATH:+:$GI_TYPELIB_PATH}"
+-  export LD_LIBRARY_PATH="@libdir@${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}"
+-  export XDG_DATA_DIRS="@datadir@${datadir:+:$datadir}"
+-else
+-  export PYTHONPATH="$(dirname $script_dir)${PYTHONPATH:+:$PYTHONPATH}"
+-  export GI_TYPELIB_PATH="$(dirname $script_dir)/libcaribou${GI_TYPELIB_PATH:+:$GI_TYPELIB_PATH}"
+-  export LD_LIBRARY_PATH="$(dirname $script_dir)/libcaribou/.libs${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}"
+-  export ANTLER_THEME_PATH="$(dirname $script_dir)/data"
+-fi
```

Comment:
The other question I have is whether or not any of this is remotely valuable in this day and age. The comments in src_prepare were
```
	# FIXME: Now this supports python3 too...
	# delete custom PYTHONPATH, useless on Gentoo and potential bug source
	# + caribou is python2 only so fix the shell scripts
#	sed -e '/export PYTHONPATH=.*python/ d' \
#		-e "s:@PYTHON@:${EPREFIX}/usr/bin/python2:" \
#		-i bin/{antler-keyboard,caribou-preferences}.in ||
#		die "sed failed"
```

Reply:
You could try with python-exec's [-native-symlinks]?

Reply:
For both `antler-keyboard.in` and `caribou-preferences.in`, the `@PYTHON@` placeholder is replaced with the fully qualified python target configured by `PYTHON_SINGLE_TARGET`, so `python-exec's [-native-symlinks]` shouldn't have an impact.

I was more questioning if any of the environment setup made sense to keep. They declares `PYTHONPATH`, `GI_TYPELIB_PATH`, `LD_LIBRARY_PATH`, and `XDG_DATA_DIRS` (the system install check passes and runs the first block). However, it looks like they are just being re-declared with the implicit global defaults.

Since everything is installed in global default locations, it seemed reasonable to remove all of the environment configuration in addition to `PYTHONPATH` (per the old ebuild comments) and further reduce "potential bug source[s]".